source/merge.ts

```diff
@@ -1,34 +1,40 @@
-import {URL, URLSearchParams} from 'url';
 import is from '@sindresorhus/is';
-import {Options, Method, NextFunction, Instance, InterfaceWithDefaults} from './utils/types';
-import knownHookEvents, {Hooks, HookType, HookEvent} from './known-hook-events';
+import {Options, Method, Defaults, NormalizedOptions, CancelableRequest, Response} from './utils/types';
+import knownHookEvents, {Hooks, HookEvent, HookType} from './known-hook-events';
+import {Got} from './create';
+import {ProxyStream} from './as-stream';
 
-export default function merge<Target extends {[key: string]: unknown}, Source extends {[key: string]: unknown}>(target: Target, ...sources: Source[]): Target & Source {
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const URLGlobal: typeof URL = typeof URL === 'undefined' ? require('url').URL : URL;
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const URLSearchParamsGlobal: typeof URLSearchParams = typeof URLSearchParams === 'undefined' ? require('url').URLSearchParams : URLSearchParams;
```

Comment:
`import {URL, URLSearchParams} from 'url';` looks better IMHO

Reply:
Encountering the same:
```
Argument of type 'URL' is not assignable to parameter of type 'string | URL'.
  Type 'URL' is not assignable to type 'import("url").URL'.
    Types of property 'searchParams' are incompatible.
      Type 'URLSearchParams' is missing the following properties from type 'URLSearchParams': entries, keys, values, [Symbol.iterator]
```